languages/C.groovy

```diff
@@ -0,0 +1,379 @@
+    MVSExec linkEdit = createLinkEditCommand(buildFile, logicalFile, member, logFile)
+
+    // execute mvs commands in a mvs job
+    MVSJob job = new MVSJob()
+    job.start()
+
+    // compile the c program
+    int rc = compile.execute()
+    int maxRC = props.getFileProperty('cc_compileMaxRC', buildFile).toInteger()
+
+    boolean bindFlag = true
+
+    if (rc > maxRC) {
+        bindFlag = false
+        String errorMsg = "*! The compile return code ($rc) for $buildFile exceeded the maximum return code allowed ($maxRC)"
+        println(errorMsg)
+        props.error = "true"
+        buildUtils.updateBuildResult(errorMsg:errorMsg,logs:["${member}.log":logFile])
+    }
+    else { // if this program needs to be link edited . . .
+
+        // Store db2 bind information as a generic property record in the BuildReport
+
+        String needsLinking = props.getFileProperty('cc_linkEdit', buildFile)
+        if (needsLinking.toBoolean()) {
+            rc = linkEdit.execute()
+            maxRC = props.getFileProperty('cc_linkEditMaxRC', buildFile).toInteger()
+
+            if (rc > maxRC) {
+                bindFlag = false
+                String errorMsg = "*! The link edit return code ($rc) for $buildFile exceeded the maximum return code allowed ($maxRC)"
+                println(errorMsg)
+                props.error = "true"
+                buildUtils.updateBuildResult(errorMsg:errorMsg,logs:["${member}.log":logFile])
+            }
+            else {
+                if(!props.userBuild){
+                    // only scan the load module if load module scanning turned on for file
+                    String scanLoadModule = props.getFileProperty('cc_scanLoadModule', buildFile)
+                    if (scanLoadModule && scanLoadModule.toBoolean())
+                        impactUtils.saveStaticLinkDependencies(buildFile, props.linkedit_loadPDS, logicalFile)
+                }
+            }
+        }
+    }
+
+    // clean up passed DD statements
+    job.stop()
+}
+
+// end script
+
+
+//********************************************************************
+//* Method definitions
+//********************************************************************
+
+/*
+ * createCParms - Builds up the c compiler parameter list from build and file properties
+ */
+def createCParms(String buildFile, LogicalFile logicalFile) {
+    def parms = props.getFileProperty('cc_compileParms', buildFile) ?: ""
+    def cics = props.getFileProperty('cc_compileCICSParms', buildFile) ?: ""
+    def sql = props.getFileProperty('cc_compileSQLParms', buildFile) ?: ""
+    def errPrefixOptions = props.getFileProperty('cc_compileErrorPrefixParms', buildFile) ?: ""
+    def compileDebugParms = props.getFileProperty('cc_compileDebugParms', buildFile)
+
+    if (buildUtils.isCICS(logicalFile))
+        parms = "$parms,$cics"
+
+    if (buildUtils.isSQL(logicalFile))
+        parms = "$parms,$sql"
+
+    if (props.errPrefix)
```

Comment:
Remote Error Feedback is only available for COBOL, PLI and Assembler. C/C++ is unfortunately not supported yet. Let's clean up the sections that leverage error feedback.